app-services/functions/writePageContents.js

```diff
@@ -39,7 +32,8 @@ async function writePageContents({ fullDocument }) {
   const updateDoc = {
     $set: {
       title,
-      doc_text: docText,
+      doc_text,
+      ...headings,
```

Comment:
can we make this a sub-document? 
```suggestion
      headings: { ...headings },
```